README.md

```diff
@@ -7,19 +7,19 @@ We used to store our artifacts in JCenter, but since [Bintray was shutdown](http
 
 In the library's main module's `build.gradle` file:
 
-```
+```groovy
 buildscript {
     repositories {
         maven { url 'https://a8c-libs.s3.amazonaws.com/android' }
     }
     dependencies {
-        classpath 'com.automattic.android:publish-to-s3:0.2'
+        classpath 'com.automattic.android:publish-to-s3:0.4.1'
```

Comment:
Note to self (just thinking / writing this out loud here (1) so I can try to make sense of it (2) so someone can confirm my reasoning or correct me if I'm wrong).

 - This `publish-to-s3` part of the `classpath` is based on the _project name_ ; see [`settings.graddle.kts`](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/update-readme-for-0.4.1/settings.gradle.kts#L1)
 - The `publish-library-to-s3` in `apply plugin:` below is defined via `plugins.register` in [`plugin/build.gradle.kts`](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/update-readme-for-0.4.1/plugin/build.gradle.kts#L34-L35)
 - The `s3PublishLibrary` below is the plugin extension, see [`project.extension.create` in `PublishLibraryToS3Plugin.kt`](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/update-readme-for-0.4.1/plugin/src/main/kotlin/com/automattic/android/PublishLibraryToS3Plugin.kt#L14)
 - The `publishLibraryToS3` gradle task name is defined [in `PublishLibraryToS3Plugin.kt` here](https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/update-readme-for-0.4.1/plugin/src/main/kotlin/com/automattic/android/PublishLibraryToS3Plugin.kt#L8)

---

@oguzkocer any reason why the extension's name doesn't seem to follow the naming logic as the project and plugin (which are all reading like "publish X to S3" but the extension reads "S3 publish X")? No idea if there is usually some naming convention around this, or if this was just arbitrary (and in that case wouldn't naming the extension `publishLibraryToS3Extension` or something like that help being consistent?)

(Not a blocker for that PR and the update of that README, just wondering)

Reply:
@AliSoftware The naming inconsistency is mostly an oversight, but also I am still not sure how to name the extensions. I don't think we'd want to have a `Extension` postfix, because in the build file the configurations are all extensions, so it'd be redundant. However, we can for sure name it `publishLibraryToS3` which would be an improvement over the existing name.

My concern is that even your suggested name doesn't make it very clear that it's about configuring the behavior of the plugin, but I guess at that point, we have to rely on convention. My uneasiness about this may be related to not being used to this convention.

Anyhow, I created an issue for taking care of this when I am making other changes to the plugin: https://github.com/Automattic/publish-to-s3-gradle-plugin/issues/21.

Thank you for the suggestion!

Reply:
> This publish-to-s3 part of the classpath is based on the project name ; see settings.graddle.kts

Although the statement is correct, the link is to the root `settings.gradle.kts` file which is incorrect. Here is where the project name is defined: https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/settings.gradle.kts#L1

When we publish an artifact, unless it's modified, it uses the `group:projectName` format where group is defined here: https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/build.gradle.kts#L12

Reply:
> Although the statement is correct, the link is to the root settings.gradle.kts file which is incorrect.

Ah, good catch! (I think that's the one I intended to link at first but messed up when changing the branch reference in the URL during pasting)

> When we publish an artifact, unless it's modified, it uses the group:projectName format where group is defined here: https://github.com/Automattic/publish-to-s3-gradle-plugin/blob/trunk/plugin/build.gradle.kts#L12

That's good info, thanks!